pkg/reconciler/workload/namespace/namespace_reconcile.go

```diff
@@ -18,62 +18,23 @@ package namespace
 
 import (
 	"context"
-	"time"
-
-	"github.com/kcp-dev/logicalcluster/v3"
 
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/labels"
 	utilserrors "k8s.io/apimachinery/pkg/util/errors"
-
-	schedulingv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/scheduling/v1alpha1"
 )
 
-type reconcileStatus int
-
-const (
-	reconcileStatusStop reconcileStatus = iota
-	reconcileStatusContinue
-)
-
-type reconciler interface {
-	reconcile(ctx context.Context, ns *corev1.Namespace) (reconcileStatus, *corev1.Namespace, error)
-}
-
-func (c *controller) reconcile(ctx context.Context, ns *corev1.Namespace) error {
-	reconcilers := []reconciler{
-		&bindNamespaceReconciler{
-			listPlacement:  c.listPlacement,
-			patchNamespace: c.patchNamespace,
-		},
-		&placementSchedulingReconciler{
-			listPlacement:  c.listPlacement,
-			enqueueAfter:   c.enqueueAfter,
-			patchNamespace: c.patchNamespace,
-			now:            time.Now,
-		},
-		&statusConditionReconciler{
-			patchNamespace: c.patchNamespace,
-		},
-	}
-
+func (c *controller) reconcile(ctx context.Context, key string, ns *corev1.Namespace) error {
 	var errs []error
 
-	for _, r := range reconcilers {
-		var err error
-		var status reconcileStatus
-		status, ns, err = r.reconcile(ctx, ns)
-		if err != nil {
-			errs = append(errs, err)
-		}
-		if status == reconcileStatusStop {
-			break
-		}
+	if err := c.reconcilePlacementBind(ctx, ns); err != nil {
+		errs = append(errs, err)
+	}
+	if err := c.reconcileScheduling(ctx, key, ns); err != nil {
+		errs = append(errs, err)
+	}
+	if err := c.reconcileStatus(ctx, ns); err != nil {
+		errs = append(errs, err)
```

Comment:
how does this block now cope with changes to spec and status?

Reply:
See the comment in committer - if both change, the status changes are not sent